src/transformers/utils/fx.py

```diff
@@ -538,11 +538,27 @@ class HFProxy(Proxy):
     """
 
     def install_metadata(self, metadata):
+        if isinstance(metadata, torch.fx.Proxy):
+            raise ValueError(
+                f"Invalid value, got torch.fx.Proxy as metadata for {self}. Expected concrete value and instead got"
+                f" {metadata}"
+            )
         self._metadata = metadata
 
     @property
     def shape(self):
-        return self.tracer.create_proxy("call_method", "size", (self,), {})
+        return self.size()
+
+    def size(self, dim=None):
```

Comment:
Question: can this be removed because `__iter__` solves also the issue?
If so, let's keep `shape` and remove `size`?

Reply:
Excellent this works.